eng/pipelines/templates/variables/image.yml

```diff
@@ -15,7 +15,7 @@ variables:
   - name: WINDOWSVMIMAGE
     value: azsdk-pool-mms-win-2022-1espt
   - name: MACVMIMAGE
-    value: macOS-12
+    value: macOS-14
```

Comment:
all the other repos we went with macos-latest we could try that here as well but I know given this repo is targeting ios it might need more control.

Reply:
No worries. I wanted to see what was breaking with the update. I do indeed need to update the `xcode` references. When I do that I'll make the update for `macos-latest`.

Reply:
I've updated to macos-latest, but I think I'm going to pin to macos-14. The reason for this is that the `XcodeVersion` changes according to which version of macOS is running. There's no map, we just have to [look it up.](https://github.com/actions/runner-images/blob/main/images/macos/macos-14-Readme.md#tools)